bitnami/redis-cluster/templates/svc-cluster-external-access.yaml

```diff
@@ -18,13 +18,32 @@ metadata:
   namespace: {{ $.Release.Namespace | quote }}
   labels: {{- include "common.labels.standard" ( dict "customLabels" $root.Values.commonLabels "context" $ ) | nindent 4 }}
     pod: {{ $targetPod }}
-  {{- if or $root.Values.cluster.externalAccess.service.annotations $root.Values.commonAnnotations }}
-  {{- $annotations := include "common.tplvalues.merge" ( dict "values" ( list $root.Values.cluster.externalAccess.service.annotations $root.Values.commonAnnotations ) "context" $ ) }}
+  {{- if or
+    ($root.Values.cluster.externalAccess.service.annotations)
+    ($root.Values.commonAnnotations)
+    (ne $root.Values.cluster.externalAccess.service.loadBalancerIPAnnotaion "") }}
+  {{- $loadBalancerIPAnnotaion := "" }}
+  {{- if ne $root.Values.cluster.externalAccess.service.loadBalancerIPAnnotaion ""}}
+    {{- $loadBalancerIPAnnotaion = printf
+      "%s: %s"
+      $root.Values.cluster.externalAccess.service.loadBalancerIPAnnotaion
+      (index $root.Values.cluster.externalAccess.service.loadBalancerIP $i) }}
+  {{- end }}
```

Comment:
I do not totally understand this part, could you try to explain it to me?

Reply:
If cluster.externalAccess.service.loadBalancerIPAnnotaion is set e.g to "metallb.universe.tf/loadbalancerips"
Then set the variable `$loadBalancerIPAnnotaion` to the string:
`"metallb.universe.tf/loadbalancerips: <n-th IP of the array listing all IPs>"`
Otherwise this new var will stay an empty string

This variable will then be merged with all other annotation in `common.tplvalues.merge` and later rendered in the output by `common.tplvalues.render`. In case of empty string it will be omitted